bindinfo/cache.go

```diff
@@ -144,14 +151,45 @@ func (bindCacheUpdater *BindCacheUpdater) Update(fullLoad bool) (err error) {
 	}
 	err = bindCacheUpdater.loadDiff(sql, newBc)
 	if err != nil {
-		return errors.Trace(err)
+		return err
+	}
+
+	bindCacheUpdater.globalHandle.Store(newBc)
+	return nil
+}
+
+func (bindCacheUpdater *BindCacheUpdater) updateOneBind(originSQL, bindSQL, defaultDB string, createTs, updateTs types.Time, status, charset, collation string) error {
+	bindCacheUpdater.lock.Lock()
```

Comment:
Why lock it here? All the operations here are thread safe.

Reply:
bind updater loop also also need write the buffer, so we need a lock